jctools-core/src/main/java/org/jctools/queues/BaseMpscLinkedArrayQueue.java

```diff
@@ -407,51 +452,11 @@ private E newBufferPeek(E[] nextBuffer, long index)
     private long newBufferAndOffset(E[] nextBuffer, long index)
     {
         consumerBuffer = nextBuffer;
-        consumerMask = (nextBuffer.length - 2) << 1;
+        consumerMask = (length(nextBuffer) - 2) << 1;
```

Comment:
This is defined in LinkedArrayQueueUtil.length as a sneaky way of simplying the generation process. i.e. for atomics all it has to do is import LinkedArrayAtomicQueueUtil.length and this will "just work"(tm)